src/commands/reword.rs

```diff
@@ -0,0 +1,354 @@
+use crate::core::dag::{resolve_commits, CommitSet, Dag, ResolveCommitsResult};
+use crate::core::effects::Effects;
+use crate::core::eventlog::{EventLogDb, EventReplayer};
+use crate::core::formatting::{printable_styled_string, Glyphs};
+use crate::core::node_descriptors::{render_node_descriptors, CommitOidDescriptor, NodeObject};
+use crate::core::rewrite::{
+    execute_rebase_plan,
+    BuildRebasePlanOptions,
+    ExecuteRebasePlanOptions,
+    // BuildRebasePlanError,
+    ExecuteRebasePlanResult,
+    RebasePlanBuilder,
+    RepoResource,
+};
+use crate::git::{CheckOutCommitOptions, Commit, GitRunInfo, MaybeZeroOid, NonZeroOid, Repo};
+
+/// Reword a commit and restack it's descendants.
+#[instrument]
+pub fn reword(
+    effects: &Effects,
+    hashes: Vec<String>,
+    messages: Vec<String>,
+    git_run_info: &GitRunInfo,
+) -> eyre::Result<isize> {
+    let repo = Repo::from_current_dir()?;
+    let references_snapshot = repo.get_references_snapshot()?;
+    let conn = repo.get_db_conn()?;
+    let event_log_db = EventLogDb::new(&conn)?;
+    let event_replayer = EventReplayer::from_event_log_db(effects, &repo, &event_log_db)?;
+    let event_cursor = event_replayer.make_default_cursor();
+    let mut dag = Dag::open_and_sync(
+        effects,
+        &repo,
+        &event_replayer,
+        event_cursor,
+        &references_snapshot,
+    )?;
+
+    let commits = match resolve_commits_from_hashes(&repo, &mut dag, effects, hashes)? {
+        Some(commits) => commits,
+        None => return Ok(1),
+    };
+
+    let messages = match build_messages(&messages, &commits, get_comment_char(&repo)?)? {
+        BuildRewordMessageResult::Succeeded { messages } => messages,
+        BuildRewordMessageResult::IdenticalMessage => {
+            writeln!(
+                effects.get_output_stream(),
+                "The message wasn't edited; nothing to do. Stopping."
+            )?;
+            return Ok(1);
+        }
+        BuildRewordMessageResult::EmptyMessage => {
+            writeln!(
+                effects.get_error_stream(),
+                "Error: the reworded message was empty. Stopping."
+            )?;
+            return Ok(1);
+        }
+    };
+
+    let subtree_roots = find_subtree_roots(&repo, &dag, &commits)?;
+
+    let rebase_plan = {
+        let pool = ThreadPoolBuilder::new().build()?;
+        let repo_pool = RepoResource::new_pool(&repo)?;
+        let mut builder = RebasePlanBuilder::new(&dag);
+
+        for root_commit in subtree_roots {
+            let only_parent_id = root_commit.get_only_parent().map(|parent| parent.get_oid());
+            if only_parent_id == None {
+                eyre::bail!(
+                    "###
+                    Refusing to reword commit {}, which has {} parents.
+                    Rewording only supported for commits with 1 parent.
+                    Stopping; nothing reworded.
+                    ###",
```

Comment:
So, some of my error messages are kind of long. Am I being too verbose w/ these?

Reply:
No, I think these error messages are quite good. (They adhere to Toulmin-style argumentation as suggested in https://www.microsoft.com/en-us/research/publication/how-should-compilers-explain-problems-to-developers/)

Stylistically, we use `Aborting` everywhere else in the codebase so far, so let's be consistent about that. (I think you could argue that `Aborting.` would be enough here, and the `nothing reworded` would be implied. It's up to you if you want to remove that second bit or not.)

```suggestion
                    "###
                    Refusing to reword commit {}, which has {} parents.
                    Rewording is only supported for commits with 1 parent.
                    Aborting; nothing reworded.
                    ###",
```

Reply:
Also, let's write this message to `effects.get_error_stream()` and return `1` as the exit code. `eyre::bail!` should be reserved for programming errors.

Reply:
OK: "stopping" -> "aborting" and I changed this to use the error stream instead of `bail!`.